code/modules/mob/living/silicon/robot/drone/drone_items.dm

```diff
@@ -203,67 +355,61 @@
-		else if(istype(W,/obj/effect/spider/spiderling))
-			stored_comms["wood"]++
-			stored_comms["wood"]++
-			stored_comms["plastic"]++
-			stored_comms["plastic"]++
-		else if(istype(W,/obj/item/weapon/light))
+		if (istype(W,/obj/item/organ))
+			playsound(src.loc, 'sound/effects/gib.ogg', 30, 1, -2)
+			src.loc.visible_message("\red [src.loc] sucks [W] into its decompiler. There's a horrible squelching noise.","\red You suck [W] into your decompiler. It makes a series of visceral crunching noises.")
+			store_buffer["biomass"] += 25
+		else if (istype(W,/obj/item/weapon/cigbutt))
+			store_buffer["plastic"] += 1
+			store_buffer["biomass"] += 1
+		else if (istype(W,/obj/effect/spider/spiderling))
+			store_buffer["biomass"] += 10
+		else if (istype(W,/obj/item/weapon/light))
 			var/obj/item/weapon/light/L = W
 			if(L.status >= 2) //In before someone changes the inexplicably local defines. ~ Z
-				stored_comms["metal"]++
-				stored_comms["glass"]++
+				store_buffer["metal"] += 5
+				store_buffer["glass"] += 5
 			else
 				continue
 		else if(istype(W,/obj/effect/decal/remains/robot))
-			stored_comms["metal"]++
-			stored_comms["metal"]++
-			stored_comms["plastic"]++
-			stored_comms["plastic"]++
-			stored_comms["glass"]++
-		else if(istype(W,/obj/item/trash))
-			stored_comms["metal"]++
-			stored_comms["plastic"]++
-			stored_comms["plastic"]++
-		else if(istype(W,/obj/effect/decal/cleanable/blood/gibs/robot))
-			stored_comms["metal"]++
-			stored_comms["metal"]++
-			stored_comms["glass"]++
-			stored_comms["glass"]++
-		else if(istype(W,/obj/item/ammo_casing))
-			stored_comms["metal"]++
-		else if(istype(W,/obj/item/weapon/shard/shrapnel))
-			stored_comms["metal"]++
-			stored_comms["metal"]++
-			stored_comms["metal"]++
-		else if(istype(W,/obj/item/weapon/shard))
-			stored_comms["glass"]++
-			stored_comms["glass"]++
-			stored_comms["glass"]++
-		else if(istype(W,/obj/item/weapon/reagent_containers/food/snacks/grown))
-			stored_comms["wood"]++
-			stored_comms["wood"]++
-			stored_comms["wood"]++
-			stored_comms["wood"]++
-		else if(istype(W,/obj/item/weapon/broken_bottle))
-			stored_comms["glass"]++
-			stored_comms["glass"]++
-			stored_comms["glass"]++
+			store_buffer["metal"] += 5
+			store_buffer["plastic"] += 5
+			store_buffer["glass"] += 5
+		else if (istype(W,/obj/item/trash))
+			store_buffer["metal"] += 1
+			store_buffer["plastic"] += 1
+		else if (istype(W,/obj/effect/decal/cleanable/blood/gibs/robot))
+			store_buffer["metal"] += 2
+			store_buffer["glass"] += 2
+		else if (istype(W,/obj/item/ammo_casing))
+			store_buffer["metal"] += 1
+		else if (istype(W,/obj/item/weapon/shard/shrapnel))
+			store_buffer["metal"] += 5
+		else if (istype(W,/obj/item/weapon/shard))
+			store_buffer["glass"] += 1
+		else if (istype(W,/obj/item/weapon/reagent_containers/food/snacks/grown))
+			store_buffer["biomass"] += 5
+		else if (istype(W,/obj/item/weapon/broken_bottle))
+			store_buffer["glass"] += 5
```

Comment:
Adapted from existing Decompiler code which was arranged in pretty much the same way.